DQM/Physics/src/SMPDQM.cc

```diff
@@ -0,0 +1,299 @@
+  if (!iEvent.getByToken(jets_, jets)) {
+    return;
+  }
+
+  for (edm::View<reco::PFJet>::const_iterator jet = jets->begin(); jet != jets->end(); ++jet) {
+    if (jet->pt() < 15.0)
+      continue;
+    TLorentzVector ijet;
+    ijet.SetPtEtaPhiM(jet->pt(), jet->eta(), jet->phi(), jet->mass());
+    recoPFJets.push_back(ijet);
+  }
+
+  std::sort(recoPFJets.begin(), recoPFJets.end(), SortByPt());
+  std::sort(selected_lep.begin(), selected_lep.end(), SortByPt());
+  if (verbose)
+    std::cout << "size of uncleaned jets" << recoPFJets.size() << std::endl;
+
+  for (unsigned int i = 0; i < recoPFJets.size(); i++) {
+    bool goodjet = false;
+    for (unsigned int j = 0; j < selected_lep.size(); j++) {
+      if (recoPFJets[i].DeltaR(selected_lep[j]) > 0.4) {
+        goodjet = true;
+        continue;
+      } else {
+        goodjet = false;
+        if (verbose)
+          std::cout << "jet\t"
+                    << "\t matched to lep with deltaR = \t" << recoPFJets[i].DeltaR(selected_lep[j]) << "\t"
+                    << selected_lep[j].Pt() << "\t" << recoPFJets[i].Pt() << std::endl;
+        break;
+      }
+    }
+    if (goodjet) {
+      TLorentzVector temp;
+      temp.Clear();
+      temp.SetPtEtaPhiM(recoPFJets[i].Pt(), recoPFJets[i].Eta(), recoPFJets[i].Phi(), recoPFJets[i].M());
+      selected_recoPFJets.push_back(temp);
+    }
+  }
+  std::cout << "size of selected jets" << selected_recoPFJets.size() << std::endl;
+  std::sort(selected_recoPFJets.begin(), selected_recoPFJets.end(), SortByPt());  // for safety
+  int njet = 0;
+  for (unsigned int k = 0; k < selected_recoPFJets.size(); k++) {
+    if (k > 4)
+      break;
+    else {
+      njet++;
+      PFJetpt->Fill(selected_recoPFJets.at(k).Pt());
+      PFJeteta->Fill(selected_recoPFJets.at(k).Eta());
+      PFJetphi->Fill(selected_recoPFJets.at(k).Phi());
+      PFJetRapidity->Fill(selected_recoPFJets.at(k).Rapidity());
+    }
+  }
+  PFJetMulti->Fill(njet);
+
+  // now we have selected jet and lepton collections
+
+  if (selected_lep.size() > 1) {
+    detall->Fill(selected_lep[0].Eta() - selected_lep[1].Eta());
+    dphill->Fill(selected_lep[0].DeltaPhi(selected_lep[1]));
+    mll->Fill((selected_lep[0] + selected_lep[1]).M());
+    ptll->Fill((selected_lep[0] + selected_lep[1]).Pt());
+    etall->Fill((selected_lep[0] + selected_lep[1]).Eta());
+    if (!selected_recoPFJets.empty()) {
+      dphi_lep1jet1->Fill(selected_recoPFJets[0].DeltaPhi(selected_lep[0]));
+      dphi_lep2jet1->Fill(selected_recoPFJets[0].DeltaPhi(selected_lep[1]));
+    }
+  }
+
+  else if (selected_lep.size() == 1) {
+    dphi_lepMET->Fill(selected_lep[0].DeltaPhi(imet));
+    mass_lepMET->Fill((selected_lep[0] + imet).M());
+    pt_lepMET->Fill((selected_lep[0] + imet).Pt());
+    if (!selected_recoPFJets.empty()) {
+      dphi_lepjet1->Fill(selected_recoPFJets[0].DeltaPhi(selected_lep[0]));
+    }
+  }  // W case
+
+  else {
+    std::cout << "zero lepton case" << endl;
```

Comment:
@anmehta in this case cout is not even protected by the ```verbose``` flag

Reply:
@fabiocos done. I removed the cout statements altogether.